test/cppsuite/tests/bounded_cursor_stress.cpp

```diff
@@ -681,6 +564,141 @@ class bounded_cursor_stress : public test {
+                if (normal_ret == 0 && exact > 0)
+                    normal_ret = normal_cursor->prev(normal_cursor.get());
+            } else
+                normal_ret = normal_cursor->prev(normal_cursor.get());
+        }
+
+        if (normal_ret == WT_NOTFOUND) {
+            testutil_assert(range_ret == WT_NOTFOUND);
+            return 0;
+        }
+        if (normal_ret != 0)
+            return normal_ret;
+        if (range_ret != 0 && range_ret != WT_NOTFOUND)
+            return range_ret;
+
+        const char *normal_key;
+        testutil_check(normal_cursor->get_key(normal_cursor.get(), &normal_key));
+        /*
+         * It is possible that there are no keys within the range. Therefore make sure that normal
+         * cursor returns a key that is outside of the range.
+         */
+        if (range_ret == WT_NOTFOUND) {
+            if (next) {
+                testutil_assert(!upper_key.empty());
+                testutil_assert(!custom_lexicographical_compare(normal_key, upper_key, true));
+            } else {
+                testutil_assert(!lower_key.empty());
+                testutil_assert(custom_lexicographical_compare(normal_key, lower_key, false));
+            }
+            return 0;
+        }
+
+        testutil_assert(range_ret == 0 && normal_ret == 0);
+
+        /* Retrieve the key the cursor is pointing at. */
+        const char *range_key;
+        testutil_check(normal_cursor->get_key(normal_cursor.get(), &normal_key));
+        testutil_check(bounded_cursor->get_key(bounded_cursor.get(), &range_key));
+        testutil_assert(std::string(normal_key).compare(range_key) == 0);
+
+        return cursor_traversal_walk(bounded_cursor, normal_cursor, lower_bound, upper_bound, next);
+    }
+
+    /*
+     * Walk both normal cursor and bounded cursor to the end of their ranges.
+     */
+    int
+    cursor_traversal_walk(scoped_cursor &bounded_cursor, scoped_cursor &normal_cursor,
+      const bound &lower_bound, const bound &upper_bound, bool next)
+    {
+        while (true) {
+            int normal_ret, range_ret;
+
+            if (next) {
+                normal_ret = normal_cursor->next(normal_cursor.get());
+                range_ret = bounded_cursor->next(bounded_cursor.get());
+            } else {
+                normal_ret = normal_cursor->prev(normal_cursor.get());
+                range_ret = bounded_cursor->prev(bounded_cursor.get());
+            }
+
+            if (normal_ret != 0 && normal_ret != WT_NOTFOUND)
+                return normal_ret;
+
+            if (range_ret != 0 && range_ret != WT_NOTFOUND)
+                return range_ret;
+
+            char *normal_key, *range_key;
+            const std::string &lower_key = lower_bound.get_key();
+            const std::string &upper_key = upper_bound.get_key();
+            /* Early exit if we have reached the end of the table. */
+            if (range_ret == WT_NOTFOUND && normal_ret == WT_NOTFOUND)
```

Comment:
we should add an assert to check that it is not possible to have `range_ret == WT_NOTFOUND && normal_ret = 0` I should have added this check earlier my mistake

Reply:
That case is checked on line 676 am I missing something?